### PR TITLE
KCL-6914 Plus button click handler

### PR DESCRIPTION
**Pull request by vladislavb-kentico**
- Implement initial iframe message for plus button
- Implement action iframe message for plus button
- Implement additional popover logic for plus button
- Refactored SDK configuration structure
- Implement custom Logger

### CODE REVIEW

**Review by PeterSkoda — CHANGES_REQUESTED** [review]
Some comments, otherwise good job!

File: src/lib/IFrameCommunicatorTypes.ts [hunk 1]
```diff
@@ -53,32 +57,55 @@ export interface IPlusRequestMessageData {
...
+  Unknown = 'Unknown',
+}
+
+export enum PlusButtonPermission {
+  CreateNew = 'CreateNew',
+  Edit = 'Edit',
+  ViewParent = 'ViewParent',
+}
+
+export enum PlusButtonPermissionOption {
```
[the remark this answers is not in the document]
Comment (ChristopherJennings) [reply src/lib/IFrameCommunicatorTypes.ts:82]: Is this a Permission status maybe?

File: src/lib/NodeSmartLinkProvider.ts [hunk 2]
```diff
@@ -240,27 +239,97 @@ export class NodeSmartLinkProvider {
...
+    const { data, targetNode } = eventData;
+
+    const messageData: DeepPartial<IPlusRequestMessageData> = {
+      ...data,
+      languageCodename: data.languageCodename ?? this.configurationManager.defaultLanguageCodename,
+      projectId: data.projectId ?? this.configurationManager.defaultProjectId,
+    };
+
+    if (validatePlusRequestMessageData(messageData)) {
+      if (isInsideIFrame()) {
```
[the remark this answers is not in the document]
Comment (vladislavb-kentico) [reply src/lib/NodeSmartLinkProvider.ts:290]: You are right, iframeCommunicator check if not necessary on l.261.

File: src/lib/IFrameCommunicatorTypes.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vladislavb-kentico) [reply src/lib/IFrameCommunicatorTypes.ts]: I decided to use `PlusButtonPermissionCheckResult` here because we already use it on the backend.

**Review by PeterSkoda — APPROVED** [review]

**Pull request merged**